packages/fuels-test-helpers/src/signers.rs

```diff
@@ -1,19 +1,24 @@
 use crate::{setup_coins, setup_test_client, wallets_config::WalletsConfig};
 use fuel_core::model::Coin;
+use fuel_core::service::Config;
 use fuel_tx::UtxoId;
 use fuels_signers::provider::Provider;
 use fuels_signers::{LocalWallet, Signer};
 use std::net::SocketAddr;
 
 #[cfg(feature = "fuels-signers")]
-pub async fn launch_provider_and_get_single_wallet() -> LocalWallet {
-    let mut wallets = launch_provider_and_get_wallets(WalletsConfig::new_single(None, None)).await;
+pub async fn launch_provider_and_get_single_wallet(node_config: Config) -> LocalWallet {
```

Comment:
since `Config::local_node()` is used throughout / most of the tests, why not make `node_config` optional and provide a default value instead of passing `Config::local_node()` throughout the codebase?

Or (maybe) even better why not make a separate function that launches the provider with configurable node?

Reply:
In new commit I added Launcher in which we can configure everything we need and in this way I tried to avoid modifying existing functions.

Reply:
Agree with the above. This covers a special case but we get more boilerplate for the common case.

Reply:
Also agree with @leviathanbeak. 

From the first principles, what we want the most is the easiest/simplest dev experience we can come up with. 

The majority of use cases will want to launch a node/provider with default configs. Some users will want to configure that themselves.

So, we're forcing the majority to go from:

```Rust
let wallet = launch_provider_and_get_single_wallet().await;
```

To: 

```Rust
let launcher = Launcher::default_config();
let wallet = launcher.launch_provider_and_get_single_wallet().await;
```

That's not very friendly (compared to the previous one). 

In addition to that, `Launcher` adds one more concept that the user has to hold in their head when working with the SDK, which is not necessarily bad, but it's better if it could be avoided and the user kept needing _fewer_ things to get the work done.

So I think the most user-friendly solution would actually be the simplest solution here: keep `launch_provider_and_get_single_wallet()` as-is and add a new one (e.g. `launch_custom_provider_and_get_single_wallet()`) that takes the config.